assembly/src/assembler/instruction/field_ops.rs

```diff
@@ -413,17 +413,13 @@ fn split_elements(span: &mut SpanBuilder) {
 ///
 /// The resulting stack after this operation is: [eq_flag, lt_flag, ...].
 ///
-/// This operation takes 6 cycles.
+/// This operation takes 3 cycles.
 fn check_lt_and_eq(span: &mut SpanBuilder) {
     // calculate a - b
     // stack: [b, a, ...] => [underflow_flag, result, ...]
     span.push_op(U32sub);
-    // Put 1 on the stack if the underflow flag was not set (there was no underflow)
-    span.push_ops([Dup0, Not]);
-    // move the result to the top of the stack and check if it was zero
-    span.push_ops([MovUp2, Eqz]);
-    // set the equality flag to 1 if there was no underflow and the result was zero
-    span.push_op(And);
+    // set the equality flag to 1 if the result was zero
+    span.push_ops([Swap, Eqz]);
```

Comment:
I noticed that we can optimize the equality check here. 

Previously we were checking whether the underflow flag is `0` _and_ the result is `0`. 
The idea is that there is no point to check the underflow flag, because the case where result is `0` and the underflow flag is `1` is impossible. 

This optimization will reduce the number of cycles by 3 for the `lt`, `lte`, `gt` and `gte` instructions.

Reply:
Nice! Could you add a more detailed comment above this line explaining this? I'd be more detailed rather than less. For example:
```
after the u32sub operation we can be in one of 3 states:
- [1, result > 0] - this means ...
- [0, result > 0] - this means ...
- [0, 0]  - this means ...

The situation of `[1, 0]` is impossible because ...
```